src/main/java/seedu/algobase/logic/commands/CommandResult.java

```diff
@@ -17,21 +17,24 @@
     /** The application should exit. */
     private final boolean exit;
 
+    private final boolean isRewind;
+
     /**
      * Constructs a {@code CommandResult} with the specified fields.
      */
-    public CommandResult(String feedbackToUser, boolean showHelp, boolean exit) {
+    public CommandResult(String feedbackToUser, boolean showHelp, boolean exit, boolean isRewind) {
         this.feedbackToUser = requireNonNull(feedbackToUser);
         this.showHelp = showHelp;
         this.exit = exit;
+        this.isRewind = isRewind;
     }
 
     /**
      * Constructs a {@code CommandResult} with the specified {@code feedbackToUser},
      * and other fields set to their default value.
      */
     public CommandResult(String feedbackToUser) {
-        this(feedbackToUser, false, false);
+        this(feedbackToUser, false, false, false);
```

Comment:
You can consider using either default values or `private static final` values to make it more obvious what the booleans mean instead